flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -212,6 +271,86 @@ public void shutdown(Time timeout) {
 				executor);
 	}
 
+	private interface RequestProcessor<T> {
+		T createRequest(HttpRequest request, ByteBuf jsonPayload) throws IOException;
+
+		void writeRequest(T body, Channel channel) throws IOException;
+	}
+
+	private static final class DefaultProcessor implements RequestProcessor<HttpRequest> {
```

Comment:
Rename to `SimpleRequest` which is initialized with a `HttpRequest` which it writes out in the writeRequest method.